### PR TITLE
test(protocol-designer): add end to end coverage for batch editing advanced settings for mix

### LINKED ISSUE #7561: Batch Edit Mix: E2E test coverage
This ticket is a placeholder, once we define testing criteria for Batch Edit Mix we can add it here!

This has to be divided into two -- multiselect and advanced transfer settings

### PR DESCRIPTION
**Pull request by reachojhaneha**
Overview
Writing the e2e tests for batch edit advanced settings for tmix steps

Changelog
Automation cases for batch edit transfer steps
Risk assessment
low

Closes
#7561

### CODE REVIEW

File: protocol-designer/cypress/integration/mixSettings.spec.js [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+}
+
+function openDesignTab() {
+  cy.get('button[id=NavTab_design]').click()
+  cy.get('button').contains('ok').click()
+
+  // Verify the Design Page
+  cy.get('#TitleBar_main > h1').contains('Multi select banner test protocol')
+  cy.get('#TitleBar_main > h2').contains('STARTING DECK STATE')
+  cy.get('button[id=StepCreationButton]').contains('+ Add Step')
```
Comment (IanLondon) [inline protocol-designer/cypress/integration/mixSettings.spec.js:28]: is this check for `StepCreationButton` just to make sure we're on the Design page?

File: protocol-designer/cypress/integration/mixSettings.spec.js [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+    cy.get('[id=BlowoutLocationField_dropdown]').should('have.value', 'trashId')
+    // Click on step 3 to verify the batch editing
+    cy.get('[data-test="StepItem_3"]').click()
+    cy.get('button[id="AspDispSection_settings_button_aspirate"]').click()
+
+    // Verify that trash is selected for the blowout option
+    cy.get('[id=BlowoutLocationField_dropdown]').should('have.value', 'trashId')
+  })
+
+  it('verify well-order indeterminate state', () => {
```
Comment (IanLondon) [inline protocol-designer/cypress/integration/mixSettings.spec.js:321]: this test case seems to check well-order determinate state case (where both 2 and 3 are "bottom to top; left to right") and not indeterminate state?

File: protocol-designer/cypress/integration/mixSettings.spec.js [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+    // click on well-order and change the order
+    cy.get('[id=WellOrderField_button_mix]').click({ force: true })
+    cy.get('h4').contains('Well Order')
+    cy.get('p').contains(
+      'Change the order in which the robot aspirates from the selected wells'
+    )
+    cy.get('select[name="mix_wellOrder_first"]')
+      .select('Bottom to top')
+      .should('have.value', 'b2t')
+    // Click done button to save the changes
```
Comment (IanLondon) [inline protocol-designer/cypress/integration/mixSettings.spec.js:337]: maybe also check `mix_wellOrder_second` in the same way that `mix_wellOrder_first` is checked?

**Review by IanLondon — COMMENTED** [review]
looking good!! Left a couple questions

File: protocol-designer/cypress/integration/mixSettings.spec.js (hunk 1 above)
Comment (reachojhaneha) [reply protocol-designer/cypress/integration/mixSettings.spec.js:28]: this verification step is only to make sure we are on Design Page

File: protocol-designer/cypress/integration/mixSettings.spec.js (hunk 2 above)
Comment (reachojhaneha) [reply protocol-designer/cypress/integration/mixSettings.spec.js:321]: So, step 2 is configured with bottom to top and step 1 has top to bottom(by default) , hence when i select these 2 steps mixed text is generated and cypress validates this step. In my opinion this is indeterminate state.

File: protocol-designer/cypress/integration/mixSettings.spec.js (hunk 3 above)
Comment (reachojhaneha) [reply protocol-designer/cypress/integration/mixSettings.spec.js:337]: this can be done. I will write a test around this

Comment (reachojhaneha) [reply protocol-designer/cypress/integration/mixSettings.spec.js:337]: this is part of e2e now:
cy.get('select[name="mix_wellOrder_second"]')
      .select('Left to right')
      .should('have.value', 'l2r')

File: protocol-designer/cypress/integration/mixSettings.spec.js [hunk 4]
```diff
@@ -0,0 +1,349 @@
+const isMacOSX = Cypress.platform === 'darwin'
+const invalidInput = 'abcdefghijklmnopqrstuvwxyz!@#$%^&*()<>?,-'
+const batchEditClickOptions = { [isMacOSX ? 'metaKey' : 'ctrlKey']: true }
```
Comment (shlokamin) [inline protocol-designer/cypress/integration/mixSettings.spec.js:3]: looks like these three lines are shared across other test files, can these be extracted into a shared util to reduce redundancy?

File: protocol-designer/cypress/integration/mixSettings.spec.js [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+      mimeType: 'application/json',
+      encoding: 'utf8',
+    })
+    cy.get('[data-test="ComputingSpinner"]').should('exist')
+    // wait until computation is done before proceeding, with generous timeout
+    cy.get('[data-test="ComputingSpinner"]', { timeout: 30000 }).should(
+      'not.exist'
+    )
+  })
+}
```
Comment (shlokamin) [inline protocol-designer/cypress/integration/mixSettings.spec.js:19]: this is almost identical to whats in the other test file too, maybe its worth making `importProtocol` a shared cypress command that takes in a file path to import, up to you thoughStart a reviewAdd single commentCancel

File: protocol-designer/cypress/integration/mixSettings.spec.js (hunk 1 above)
Comment (shlokamin) [reply protocol-designer/cypress/integration/mixSettings.spec.js:28]: same thing here with `openDesignTab` seems redundant across the batch edit test files

File: protocol-designer/cypress/integration/mixSettings.spec.js [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+  before(() => {
+    cy.visit('/')
+    cy.closeAnnouncementModal()
+    importProtocol()
+    openDesignTab()
+  })
+  it('Verify functionality of mix settings with different labware', () => {
+    enterBatchEdit()
+
+    // Different labware disbales aspirate and dispense Flowrate , tipPosition, delay and touchTip
```
Comment (shlokamin) [inline protocol-designer/cypress/integration/mixSettings.spec.js:46]: `disbales` 😛

File: protocol-designer/cypress/integration/mixSettings.spec.js [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+    cy.get('button').contains('Done').click()
+
+    // Click save button to save the changes
+    cy.get('button').contains('save').click()
+
+    // Click on step 1 as it has flowrate set to 100 from previous testcase
+    cy.get('[data-test="StepItem_1"]').click(batchEditClickOptions)
+
+    // indeterminate state in flowrate is empty
+    cy.get('input[name="aspirate_flowRate"]').should('have.value', '')
```
Comment (shlokamin) [inline protocol-designer/cypress/integration/mixSettings.spec.js:135]: can we also check the placeholder is a dash? https://github.com/Opentrons/opentrons/pull/7805/files#r638193856

**Review by shlokamin — COMMENTED** [review]
Nice! Left some comments but nothing major. Good stuff 😄

**Comment by shlokamin** [comment]
Also, linting is failing bcuz u have to run prettier

File: protocol-designer/cypress/integration/mixSettings.spec.js (hunk 2 above)
Comment (IanLondon) [reply protocol-designer/cypress/integration/mixSettings.spec.js:321]: ohh I get it, I wasn't looking at line 344

**Review by IanLondon — APPROVED** [review]

File: protocol-designer/cypress/integration/mixSettings.spec.js (hunk 1 above)
Comment (reachojhaneha) [reply protocol-designer/cypress/integration/mixSettings.spec.js:28]: openDesignTab uses the selectors which are not very stable, hence i have not used it. About the redundancy part, I agree. I will make a util for the import protocol. I am working on the new feature now, hence will not be making changes to the existing code.

File: protocol-designer/cypress/integration/mixSettings.spec.js (hunk 4 above)
Comment (reachojhaneha) [reply protocol-designer/cypress/integration/mixSettings.spec.js:3]: invalidInput in my opinion is a variable that will change with every file. So its better to have it test suite dependent. And about the other 2 lines of code can be put into a util. For future work.

File: protocol-designer/cypress/integration/mixSettings.spec.js (hunk 6 above)
Comment (reachojhaneha) [reply protocol-designer/cypress/integration/mixSettings.spec.js:46]: this is corrected.

**Pull request merged**